pkg/cri/server/instrumented_service.go

```diff
@@ -455,6 +1497,48 @@ func (in *instrumentedService) Version(ctx context.Context, r *runtime.VersionRe
 	return res, errdefs.ToGRPC(err)
 }
 
+func (in *instrumentedAlphaService) Version(ctx context.Context, r *runtime_alpha.VersionRequest) (res *runtime_alpha.VersionResponse, err error) {
+	if err := in.checkInitialized(); err != nil {
+		return nil, err
+	}
+	log.G(ctx).Tracef("Version with client side version %q", r.GetVersion())
+	defer func() {
+		if err != nil {
+			log.G(ctx).WithError(err).Error("Version failed")
+		} else {
+			log.G(ctx).Tracef("Version returns %+v", res)
+		}
+	}()
+	// converts request and response for earlier CRI version to call and get response from the current version
+	p, err := r.Marshal()
+	if err == nil {
+		var v1r runtime.VersionRequest
+		if err = v1r.Unmarshal(p); err == nil {
+			var v1res *runtime.VersionResponse
+			v1res, err = in.c.Version(ctrdutil.WithNamespace(ctx), &v1r)
```

Comment:
The Version is v1 now. 

```
// Version returns the runtime name, runtime version and runtime API version.
func (c *criService) Version(ctx context.Context, r *runtime.VersionRequest) (*runtime.VersionResponse, error) {
	return &runtime.VersionResponse{
		Version:           kubeAPIVersion,
		RuntimeName:       containerName,
		RuntimeVersion:    version.Version,
		RuntimeApiVersion: constants.CRIVersion,
	}, nil
}
```

We need different function for v1alpha.

Reply:
good catch I saw that when going through then forgot...

Reply:
new output when connected to the v1alpha2 cri service entry points

sudo crictl version
```
Version:  0.1.0
RuntimeName:  containerd
RuntimeVersion:  v1.5.0-170-ga5c417ac0.m
RuntimeApiVersion:  v1alpha2
```